File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def goodNodes(self, root, max_value = -10000):
        """
        :type root: TreeNode
        :rtype: int
        """
        
        return self.goodNodes(root.left, max(max_value, root.val)) + self.goodNodes(root.right, max(max_value, root.val)) + (root.val >= max_value) if root else 0

File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func getGoodNodes(root *TreeNode, prev int, count *int){
    if root == nil{
        return 
    }
    if prev <= root.Val{
        *count ++
        prev = root.Val
    }
    getGoodNodes(root.Left, prev, count)
    getGoodNodes(root.Right, prev, count)
}
func goodNodes(root *TreeNode) (goodNodeCount int)  {
    getGoodNodes(root, root.Val, &goodNodeCount)
    return 
}